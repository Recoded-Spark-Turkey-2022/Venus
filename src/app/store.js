// eslint-disable-next-line no-unused-vars
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import listingReducer from '../features/listings/listingSlice';

export const store = configureStore({
  reducer: {
    listing: listingReducer,
  },
  // eslint-disable-next-line no-shadow
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
