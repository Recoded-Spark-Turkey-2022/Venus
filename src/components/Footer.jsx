import React from 'react';
import { Link } from 'react-router-dom';
import Logo from '../assets/Logo.svg'

const Footer = ()=>{

    return(
        <footer className='flex flex-wrap flex-col md:flex-row items-center justify-center bg-white p-2'>
          <div className='flex justify-center items-center  mt-5 mb-5 md:mt-1 md:mb-1 mx-auto flex-shrink-0'>
            <img src={Logo} alt="" />
          </div>
          <div className='w-full flex-grow lg:flex lg:items-center lg:w-auto'>
            <ul className="flex justify-center">
              <li className='mr-6'><Link className='text-[#8b8f9c] hover:text-[#4699c2]' to='/'>Home</Link></li>
              <li className='mr-6'><Link className='text-[#8b8f9c] hover:text-[#4699c2]' to='/about'>About</Link></li>
              <li className='mr-6'><Link className='text-[#8b8f9c] hover:text-[#4699c2]' to='/contact'>Contact</Link></li>
              <li className='mr-6'><Link className='text-[#8b8f9c] hover:text-[#4699c2]' to='/blogs'>Blogs</Link></li>
            </ul>
          </div>
          <div className='mx-auto mt-10 sm:mt-1 px-16 sm:px-8 flex lg:justify-center'>
            <div className='flex justify-between flex-row'>
            <div className='flex justify-between items-center mr-3'>
              <button className='text-white rounded-full bg-mediumBlue px-8 ' type='button' >Sign up</button>
            </div>
            <div className=''>
              <select className='rounded-full border-[#4699c2] px-8'> 
                <option className='border-[#4699c2]' value='en'>English</option>
                <option value='ar'>Arabic</option>
              </select>
            </div>
            </div>

          </div>

        </footer>
    )
}
export default Footer;