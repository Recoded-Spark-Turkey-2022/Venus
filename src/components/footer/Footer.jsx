import React from 'react';
import { Link } from 'react-router-dom';
import Logo from '../../assets/Logo.svg';

const Footer = () => {
  return (
    <footer className=" flex   container mx-auto flex-col md:flex-row items-center justify-center bg-white py-10 px-5">
      <div className="flex justify-center items-center  mt-5 mb-5 md:mt-1 md:mb-1 mx-auto flex-shrink-0">
        <a
          href="#navbar"
          className="brightness-[70%] hover:brightness-[110%] duration-500"
        >
          <img src={Logo} alt="Refubook-logo" />
        </a>
      </div>
      <div className="w-full mb-2 md:mb-1 lg:ml-3 flex-grow lg:flex lg:items-center lg:w-auto">
        <ul className="flex gap-8 items-center font-medium justify-center">
          <li className="">
            <Link className="text-darkGrey hover:text-darkBlue" to="/">
              Home
            </Link>
          </li>
          <li className="">
            <Link className="text-darkGrey hover:text-darkBlue" to="/about">
              About
            </Link>
          </li>
          <li className="">
            <Link className="text-darkGrey hover:text-darkBlue" to="/contact">
              Contact
            </Link>
          </li>
          <li className="">
            <Link className="text-darkGrey hover:text-darkBlue" to="/blogs">
              Blogs
            </Link>
          </li>
        </ul>
      </div>
      <div className="mx-auto mt-5 sm:mt-1 flex lg:justify-center">
        <div className="flex justify-between flex-row">
          <div className="flex justify-between items-center mr-3">
            <Link to='/signup'>
            <button
              className="text-[#FFFFFF] rounded-full bg-mediumBlue px-8 btn-default transition-colors
     hover:bg-white ease-in duration-300 border border-mediumBlue  whitespace-nowrap
     hover:border-mediumBlue hover:text-mediumBlue"
              type="button"
            >
              Sign up
            </button>
            </Link>
          </div>
          <div className="">
            <select className="rounded-full border border-solid outline-none text-darkGrey font-medium border-mediumBlue px-8">
              <option className="border-mediumBlue " value="en">
                English
              </option>
              <option value="ar">Arabic</option>
            </select>
          </div>
        </div>
      </div>
    </footer>
  );
};
export default Footer;
