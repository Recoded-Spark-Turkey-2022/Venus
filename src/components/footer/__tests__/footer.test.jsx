import React from 'react';
import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import { store } from '../../../app/store';
import { BrowserRouter as Router } from "react-router-dom";
import Footer from "../Footer";

jest.mock('react-i18next', () => ({
    useTranslation: () => {
      return {
        t: (str) => str
      };
    },
  }));

it(`renders correctly` , () => {
    const tree = renderer.create(
    <Router>
      <Provider store={store}>
        <Footer />
        </Provider>
    </Router>);
    expect(tree).toMatchSnapshot();
});

