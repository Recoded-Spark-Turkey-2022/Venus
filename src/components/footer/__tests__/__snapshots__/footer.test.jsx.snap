// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  className=" relative z-10 flex   container mx-auto flex-col md:flex-row items-center justify-center bg-white py-10 px-5"
>
  <div
    className="flex justify-center items-center  mt-5 mb-5 md:mt-1 md:mb-1 mx-auto flex-shrink-0"
  >
    <a
      className="brightness-[70%] hover:brightness-[110%] duration-500"
      href="#navbar"
    >
      <img
        alt="Refubook-logo"
        src="Logo.svg"
      />
    </a>
  </div>
  <div
    className="w-full mb-2 md:mb-1 lg:ml-3 flex-grow lg:flex lg:items-center lg:w-auto"
    data-testid="f-links"
  >
    <ul
      className="flex gap-8 items-center font-medium justify-center"
    >
      <li
        className=""
      >
        <a
          className="text-darkGrey hover:text-darkBlue"
          href="/"
          onClick={[Function]}
        >
          Footer.home
        </a>
      </li>
      <li
        className=""
      >
        <a
          className="text-darkGrey hover:text-darkBlue"
          href="/about"
          onClick={[Function]}
        >
          Footer.about
        </a>
      </li>
      <li
        className=""
      >
        <a
          className="text-darkGrey hover:text-darkBlue"
          href="/contact"
          onClick={[Function]}
        >
          Footer.contact
        </a>
      </li>
      <li
        className=""
      >
        <a
          className="text-darkGrey hover:text-darkBlue"
          href="/blogs"
          onClick={[Function]}
        >
          Footer.blogs
        </a>
      </li>
    </ul>
  </div>
  <div
    className="mx-auto mt-5 sm:mt-1 flex lg:justify-center"
  >
    <div
      className="flex justify-between flex-row"
    >
      <div
        className="flex justify-between items-center mr-3"
      >
        <a
          href="/signup"
          onClick={[Function]}
        >
          <button
            className="px-8 whitespace-nowrap rounded-full"
            id="mediumBlue-button"
            type="button"
          >
            Button.su
          </button>
        </a>
      </div>
      <div
        className=""
      >
        <select
          className="rounded-full border border-solid outline-none text-darkGrey font-medium border-mediumBlue px-8"
          onChange={[Function]}
        >
          <option
            className="border-mediumBlue "
            value="en"
          >
            English
          </option>
          <option
            value="ar"
          >
            العربية
          </option>
        </select>
      </div>
    </div>
  </div>
</footer>
`;
